core/drivers/clk/clk.c

```diff
@@ -392,6 +395,12 @@ static void print_clock_subtree(struct clk *clk_root __maybe_unused,
 #ifdef CFG_DRIVERS_CLK_PRINT_TREE
 	struct clk *clk = NULL;
 
+	if (indent == CLK_PRINT_TREE_MAX_RECURSION) {
+		EMSG("Abort clk subtree loop, %d recursive calls reached",
+		     CLK_PRINT_TREE_MAX_RECURSION);
+		return;
+	}
+
 	STAILQ_FOREACH(clk, &clock_list, link) {
 		if (clk_get_parent(clk) == clk_root) {
```

Comment:
By the way, i'll also rename `clk_root` to `parent`, as suggested during the `regulator_print_tree()` implementation review.